Wikipedia/Code/TalkPageLocalHandler.swift

```diff
@@ -20,14 +20,23 @@ extension NSManagedObjectContext {
         fetchRequest.fetchLimit = 1
         fetchRequest.predicate = NSPredicate(format: "key == %@", databaseKey)
         
-        return try fetch(fetchRequest).first
+        let talkPage = try fetch(fetchRequest).first
+        talkPage?.dateAccessed = Date()
```

Comment:
I'm a bit concerned that the method for fetching a talk page also mutates it and saves the context - could I move this into a separate function that's explicitly called when the user views the talk page or is there some other consideration that I'm missing?

Reply:
I think I misread "accessed" to mean "viewed by the user" - is the intent to have this property updated anytime the object is accessed even by non-user task?

Reply:
@joewalsh true, agreed, non-user task shouldn't update this in my opinion. I really meant viewed by the user & wanted to keep track of what talk pages they seem to be interested in lately and update that date. Then later I sort by that date so we are deleting talk pages that they don't seem to be visiting anymore. To keep it simpler we could probably only timestamp this after it's fetched in the view controller so we know it was instigated by the user.

Reply:
@tonisevener makes sense. I pushed a change, but feel free to alter or revert if you had other ideas